mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from collections.abc import Iterable
+from functools import partial
+from multiprocessing import Pool
+from typing import Callable, Tuple
+
+import rich
+from rich.progress import Progress
+
+from ..utils import is_seq_of
+from .timer import Timer
+
+
+class RichProgressBar:
+    """use rich to enhance progress bar.
+
+    This class uses the rich library to enhance the progressbar,
+    Do not instantiate two RichProgressbars at the same time.
+
+    Examples:
+        >>> import mmengine
+        >>> import time
+        >>> bar = mmengine.RichProgressBar()
+        >>> bar.add_task(10)
+        >>> for i in range(10):
+        >>>     bar.update()
+        >>>     time.sleep(1)
+    """
+
+    def __init__(self):
+        self.bar = Progress()
```

Comment:
Cound RichProgressBar accept arguments like this?
```suggestion
    def __init__(self, *args, **kwargs):
        self.bar = Progress(*args, **kwargs)
```